R/ddi.R

```diff
@@ -0,0 +1,147 @@
+#' Create a Drug-Drug Interaction (DDI) Simulation
+#'
+#' @param ... a set of compounds created by `compound()`
+#'
+#' @return a Drug-Drug Interaction (DDI) simulation object
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' rifampicin <- compound("Rifampicin")
+#' midazolam <- compound("Midazolam")
+#' create_ddi(rifampicin, midazolam)
+#' }
+create_ddi <- function(...) {
+  ddi <- DDI$new()
+
+  ddi$combine(...)
+  ddi
+}
+
+#' Import Drug-Drug Interaction (DDI) Simulation from Snapshot
+#'
+#' @inheritParams get_source
+#'
+#' @return a DDI object
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' import_ddi("Rifampicin-Midazolam-DDI.json")
+#' }
+import_ddi <- function(input) {
+  ddi <- DDI$new()
+  ddi$import(input = input)
+  ddi
+}
+
+#' R6 Class Representing a DDI Snapshot
+#'
+#' @description
+#' A snapshot is a JSON file that contains all the information about a model.
+#' This class is used to import, access and manipulate the data from a DDI snapshot.
+DDI <- R6::R6Class(
+  classname = "DDI",
+  class = TRUE,
+  inherit = Snapshot,
+  public = list(
+    #' @description
+    #' Create a DDI object.
+    #' @return A new `DDI` object.
+    initialize = function() {
+      self$source <- NULL
+      self$data <- list()
+    },
+    #' @description
+    #' Combine multiple compounds into a DDI simulation.
+    #' @param ... a set of compounds created by `compound()`
+    combine = function(...) {
+      self$source <- "Merge"
+
+      snapshots <- list(...)
+
+      snapshot_versions <- map(snapshots, ~ .x$data$Version) %>% list_c()
+
+      # If Snapshots have different versions
+      if (length(unique(snapshot_versions)) > 1) {
+        cli_process_start("Multiple versions detected. Converting to the latest version.")
+
+        temp_dir <- tempfile()
+        dir.create(temp_dir)
```

Comment:
Maybe make sure the temp_dir gets cleaned at some point.

Reply:
When session is closed, all tempfile() are deleted